drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,841 @@
+}
+
+uint8_t sx127x_get_rx_single(sx127x_t *dev)
+{
+    return dev->settings.lora.rx_continuous ? false : true;
+}
+
+void sx127x_set_rx_single(sx127x_t *dev, uint8_t single)
+{
+    DEBUG("[DEBUG] Set RX single: %d\n", single);
+
+    dev->settings.lora.rx_continuous = single ? false : true;
+}
+
+bool sx127x_get_crc(sx127x_t *dev)
+{
+#if defined(MODULE_SX1272)
+    return (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1) &
+            SX1272_RF_LORA_MODEMCONFIG1_RXPAYLOADCRC_MASK);
+#else /* MODULE_SX1276 */
+    return (sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG2) &
+            SX1276_RF_LORA_MODEMCONFIG2_RXPAYLOADCRC_MASK);
+#endif
+}
+
+void sx127x_set_crc(sx127x_t *dev, bool crc)
+{
+    DEBUG("[DEBUG] Set CRC: %d\n", crc);
+
+    dev->settings.lora.crc_on = crc;
+#if defined(MODULE_SX1272)
+    uint8_t config2_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1);
+    config2_reg &= SX1272_RF_LORA_MODEMCONFIG1_RXPAYLOADCRC_MASK;
+    config2_reg |= crc << 1;
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, config2_reg);
+#else /* MODULE_SX1276 */
+    uint8_t config2_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG2);
+    config2_reg &= SX1276_RF_LORA_MODEMCONFIG2_RXPAYLOADCRC_MASK;
+    config2_reg |= crc << 2;
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG2, config2_reg);
+#endif
+}
+
+uint8_t sx127x_get_hop_period(sx127x_t *dev)
+{
+    return sx127x_reg_read(dev, SX127X_REG_LR_HOPPERIOD);
+}
+
+void sx127x_set_hop_period(sx127x_t *dev, uint8_t hop_period)
+{
+    DEBUG("[DEBUG] Set Hop period: %d\n", hop_period);
+
+    dev->settings.lora.freq_hop_period = hop_period;
+
+    uint8_t tmp = sx127x_reg_read(dev, SX127X_REG_LR_PLLHOP);
+    if (dev->settings.lora.freq_hop_on) {
+        tmp |= SX127X_RF_LORA_PLLHOP_FASTHOP_ON;
+        sx127x_reg_write(dev, SX127X_REG_LR_PLLHOP, tmp);
+        sx127x_reg_write(dev, SX127X_REG_LR_HOPPERIOD, hop_period);
+    }
+}
+
+bool  sx127x_get_fixed_header_len_mode(sx127x_t *dev)
+{
+    return dev->settings.lora.use_fix_len;
+}
+
+void sx127x_set_fixed_header_len_mode(sx127x_t *dev, bool fixed_len)
+{
+    DEBUG("[DEBUG] Set fixed header length: %d\n", fixed_len);
+
+    dev->settings.lora.use_fix_len = fixed_len;
+
+    uint8_t config1_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG1);
+#if defined(MODULE_SX1272)
+    config1_reg &= SX1272_RF_LORA_MODEMCONFIG1_IMPLICITHEADER_MASK;
+    config1_reg |= fixed_len << 2;
+#else /* MODULE_SX1276 */
+    config1_reg &= SX1276_RF_LORA_MODEMCONFIG1_IMPLICITHEADER_MASK;
+    config1_reg |= fixed_len;
+#endif
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG1, config1_reg);
+}
+
+uint8_t sx127x_get_payload_length(sx127x_t *dev)
+{
+    return dev->settings.lora.payload_len;
+}
+
+void sx127x_set_payload_length(sx127x_t *dev, uint8_t len)
+{
+    DEBUG("[DEBUG] Set payload len: %d\n", len);
+
+    if (dev->settings.lora.use_fix_len) {
+        dev->settings.lora.payload_len = len;
```

Comment:
This variable get's outdated as soon as you send something. I think storing it isn't necessary.